back-end/src/app.module.ts

```diff
@@ -8,9 +8,7 @@ import { TaskService } from './task/task.service';
 @Module({
   imports: [
     ConfigModule.forRoot(),
-    MongooseModule.forRoot(
-      `mongodb://${process.env.DB_HOST}:${process.env.DB_PORT}/${process.env.DB_NAME}`,
-    ),
+    MongooseModule.forRoot(<string>process.env.DB_URI),
```

Comment:
In this line we (and also the mongoose `forRoot` function) want to avoid `undefined` or `null` values, which in a scenario where the someone did not set the environment variables correctly, can definitely occur.

However, in the aforementioned scenario where the environment variable is undefined, with your code `<string>undefined` will still be `undefined` even after casting. This is bad, because with this method only the compile time "I only accept string" warning/error is resolved, but runtime it still can get an undefined value, that could lead to unexpected behaviour.

Reply:
Would `MongooseModule.forRoot(<string>process.env.DB_URI ?? '')` be a solution for this problem?

Reply:
`<string>undefined` will still be `undefined` , so the casting is unnecessary. Simply `(process.env.DB_URI ?? '')` is good

both reach the same goal, but if you use casting, you will have 1 more computation